### ISSUE TITLE
Bug with inventory slots (duplication glitch)

### ISSUE THREAD

**Issue by aramperes** [GlowstoneMC/Glowstone#313]
Version: 1.9.4 build 157

Steps to reproduce (**[video](http://momoperes.ca/bug2.mp4)**):
1. Place an item in the top-left slot. This will cause a duplicate of the item to appear in the off-hand slot
2. Shift-click on the top-left slot. This will bring the original item to the first slot, and create another copy of the item in the boots slot. It will also cause a server error (see stacktrace below)
3. Take the item in the boot slot and place it in another slot
4. Take the item in the off-hand slot and place it in another slot
5. Close the inventory. The original item (slot 0) will disappear, while the two duplicates will remain.

The stacktrace (step 2):

```
java.lang.NullPointerException
        at net.glowstone.inventory.GlowInventory.tryToFillSlots(GlowInventory.java:200)
        at net.glowstone.inventory.GlowPlayerInventory.handleShiftClick(GlowPlayerInventory.java:155)
        at net.glowstone.net.handler.play.inv.WindowClickHandler.process(WindowClickHandler.java:325)
        at net.glowstone.net.handler.play.inv.WindowClickHandler.handle(WindowClickHandler.java:32)
        at net.glowstone.net.handler.play.inv.WindowClickHandler.handle(WindowClickHandler.java:27)
        at net.glowstone.libs.com.flowpowered.network.session.BasicSession.handleMessage(BasicSession.java:80)
        at net.glowstone.libs.com.flowpowered.network.session.BasicSession.messageReceived(BasicSession.java:139)
        at net.glowstone.net.GlowSession.pulse(GlowSession.java:474)
        at java.util.concurrent.ConcurrentHashMap$KeySetView.forEach(Unknown Source)
        at net.glowstone.net.SessionRegistry.pulse(SessionRegistry.java:23)
        at net.glowstone.scheduler.GlowScheduler.pulse(GlowScheduler.java:144)
        at net.glowstone.scheduler.GlowScheduler.lambda$start$0(GlowScheduler.java:83)
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
        at java.util.concurrent.FutureTask.runAndReset(Unknown Source)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(Unknown Source)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
```

**Comment by mastercoms** [comment]
does this happen before the most recent inventory pr?

**Comment by aramperes** [comment]
@mastercoms Cannot confirm this bug prior to #279, seems like a problem with slot ids.

**Comment by mastercoms** [comment]
I meant #272

**Comment by aramperes** [comment]
Should be fixed in #293.

**Comment by mastercoms** [comment]
confirmed fixed.
